docs/test_development.rst

```diff
@@ -82,64 +80,90 @@ Moreover the unit tests are executed for `each new pull request via Travis
 
-Functional API tests are stored in logical chunks in files. Tests are tagged by tags.
-So it is possible to run for example just Gluster tests. All tests should have
-fixture for proper ``setup`` and ``teardown``. All objects created during testing
-should be removed after test run. There should not be any remains after test run.
+    * fixture to read markers and change import accordingly
+    * fixture to read markers and check if the cluster matches the
+      requirements (eg. do we have specified volume type there?)
+    * multiple clusters
 
-There are *only* 2 exceptions:
-* CRUD *happy path* tests which are stored in one file where they share
-  object created and deleted during testing tests from file. These tests should run
-  in same order like they are written in the file. CRUD tests **are not** tagged.
+Tagging makes it possible to run for example just tests related to particular
+volume which requires profiling to be enabled.
 
-* create or import cluster tests which run at the same beginning of testing because
-  they left created/imported clusters for further testing. This exception exists
-  because cluster creating and importing have extremly big resource needs.
+All tests should use a proper pytest fixture for setup and teardown, if setup
+or teardown is needed. All objects created during testing should be removed
+after test run. The same applies for the fixtures, if something is created
+during setup phase, it should be removed during teardown. There should not be
+any remains after test run.
 
-Files with tests example:
+Exceptions
+``````````
 
-* crud_ceph_pool.py - contents create/read/update/delete Ceph pool tests.
-  Tests should run in exact order. Tests share ``pool`` object which is
-  created in 1st test, read in 2nd test, update in 3rd test and deleted
-  in last test. This kind of tests are usually run in CI 
-  for smoke/reggression/acceptance testing. Sharing of ``pool`` object
-  saves resources. These tests are only *positive* or *happy path*.
-  All mentioned tests for one object are stored in one file
-  with prefix crud + [storage type] + tested object.
+There are only 2 exceptions from the rules listed above.
 
-* create_ceph_cluster.py - In file with name in form (create|import) + [storage type]
-  are stored *positive* or *happy path* tests for importing or creating cluster
-  for one storage type. These tests are run before any other tests because
-  they create or import cluster which is used during further testing.
-  Cluster identification is stored in configuration so it can be used in other tests.
-  Cluster is identified by one of its machine during import process. Once it is imported
-  or created it is identified by ``cluster id``. This will change once Tendrl can
-  internally identify clusters by their names.
+Test cases which test import or unamanage cluster operations itself should
+not use ``imported_cluster`` fixture, but handle the import itself in the code
+of the test case.
```

Comment:
Is there any reason not to use ``imported_cluster`` fixture for unmanage test? (Except that it will not work unmanage in the teardown of the fixture, but that might be simply skipped.)

Reply:
Ok, in this case it makes sense to use the fixture.